components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/odata/MongoDataHandler.java

```diff
@@ -0,0 +1,427 @@
+        return createdEntry;
+    }
+
+    /**
+     * This method deletes the entity from the collection for a given key.
+     *
+     * @param tableName Name of the table
+     * @param entity    Entity
+     * @throws ODataServiceFault
+     */
+    public boolean deleteEntityInTable(String tableName, ODataEntry entity) {
+
+        String keyValue = entity.getValue(DOCUMENT_ID);
+        WriteResult delete = jongo.getCollection(tableName).remove(new ObjectId(keyValue));
+        return delete.wasAcknowledged();
+    }
+
+    /**
+     * This method updates the given entity in the given collection.
+     *
+     * @param tableName     Name of the table
+     * @param newProperties New Properties
+     * @throws ODataServiceFault
+     */
+    public boolean updateEntityInTable(String tableName, ODataEntry newProperties) {
+
+        List<String> pKeys = this.primaryKeys.get(tableName);
+        String newPropertyObjectKeyValue = null;
+        for (String newPropertyObjectKeyName : newProperties.getData().keySet()) {
+            if (newPropertyObjectKeyName.equals(DOCUMENT_ID)) {
+                newPropertyObjectKeyValue = newProperties.getValue(newPropertyObjectKeyName);
+            }
+        }
+        for (String column : newProperties.getData().keySet()) {
+            if (!pKeys.contains(column)) {
+                String propertyValue = newProperties.getValue(column);
+                assert newPropertyObjectKeyValue != null;
+                jongo.getCollection(tableName).update(new ObjectId(newPropertyObjectKeyValue)).
+                    with(SET + column + ": '" + propertyValue + "'}}");
+            }
+        }
+        return true;
+    }
+
+    /**
+     * This method updates the entity in table when transactional update is necessary.
+     *
+     * @param tableName     Table Name
+     * @param oldProperties Old Properties
+     * @param newProperties New Properties
+     * @throws ODataServiceFault
+     */
+    public boolean updateEntityInTableTransactional(String tableName, ODataEntry oldProperties, ODataEntry newProperties) {
+
+        List<String> pKeys = this.primaryKeys.get(tableName);
+        String newPropertyObjectKeyValue = null;
+        String oldPropertyObjectKeyValue = null;
+        for (String newPropertyObjectKeyName : newProperties.getData().keySet()) {
+            if (newPropertyObjectKeyName.equals(DOCUMENT_ID)) {
+                newPropertyObjectKeyValue = newProperties.getValue(newPropertyObjectKeyName);
+            }
+        }
+        for (String oldPropertyObjectKeyName : oldProperties.getData().keySet()) {
+            if (oldPropertyObjectKeyName.equals(DOCUMENT_ID)) {
+                oldPropertyObjectKeyValue = oldProperties.getValue(oldPropertyObjectKeyName);
+            }
+        }
+        for (String column : newProperties.getData().keySet()) {
+            if (!pKeys.contains(column)) {
```

Comment:
Can pKey null? If so should handle NPE

Reply:
No. pKey can never be null. _id is the default primary key in MongoDB.